cirq/contrib/qasm_import/_parser.py

```diff
@@ -71,21 +82,23 @@ def on(self, args: List[List[cirq.Qid]],
         # used as arguments, we generate reg_size GateOperations via iterating
         # through each qubit of the registers 0 to n-1 and use the same one
         # qubit from the "single-qubit registers" for each operation.
-        for i in range(reg_size):
-            qubits = []  # type: List[cirq.Qid]
-            for qreg in args:
-                if len(qreg) == 1:  # single qubits
-                    qubit = qreg[0]
-                else:  # reg_size size register
-                    qubit = qreg[i]
-                if qubit in qubits:
-                    raise QasmException("Overlapping qubits in arguments"
-                                        " at line {}".format(lineno))
-                qubits.append(qubit)
+        op_qubits = functools.reduce(np.broadcast, args)  # type: np.broadcast
+        for qubits in op_qubits:  # type: Tuple[cirq.Qid]
+            if len(np.unique(qubits)) < len(qubits):
+                raise QasmException("Overlapping qubits in arguments"
+                                    " at line {}".format(lineno))
             yield self.cirq_gate.on(*qubits)
 
 
 class QasmParser:
+    """Parser for QASM strings.
+
+        Example:
+
+         >>> qasm = "OPENQASM 2.0; qreg q1[2]; CX q1[0], q1[1];"
```

Comment:
Don't indent `Example:`. Don't include the `>>>` before code; just indent it 4 spaces.

Reply:
ack